test/Constraints/subscript.swift

```diff
@@ -216,3 +216,13 @@ func rdar61084565() {
   let a = Foo()
   a[] // expected-error {{value of type 'Foo' has no subscripts}}
 }
+
+func test_subscript_accepts_type_name_argument() {
+  struct A {
+    subscript(a: A.Type) -> Int { get { 42 } }
+  }
+
+  func test(a: A?) {
+    let _ = a?[A] // Ok
```

Comment:
Did we accept this by mistake? I guess we should at least warn about the missing “.self” instead of making it an error

Reply:
Yes, I think it was just an oversight but fixing a crash make this fail to type-check suggesting `.self` which is a source compatibility regression unfortunately... I'll try to figure out how to make it a warning in MiscDiagnostics but for that I'd need access to whole parent AST and not just an immediate parent expr...

Reply:
Once revert of a revert lands I'm going to adjust this to report an error with a fix-it.

Reply:
Revert has been merged and test-case is now going to fail with fix-its.